### ISSUE TITLE
Some activities from mastodon.social are missing an object_uri?

### ISSUE THREAD

**Issue by kescherCode** [mastodon/mastodon#19511]
### Steps to reproduce the problem

1. Check "Dead" tab on Sidekiq
2. See dead entries with error "NoMethodError: undefined method `start_with?' for nil:NilClass"
...

### Expected behaviour

These entries should not exist

### Actual behaviour

They do exist, indicating a failure either on the sending side or the receiving side

### Detailed description

I only see them on activities from mastodon.social.

ActivityPub::ProcessingWorker – NoMethodError: undefined method `start_with?' for nil:NilClass

```
Arguments: 108148003666047435, "{\"@context\":[\"https://www.w3.org/ns/activitystreams\",{\"Hashtag\":\"as:Hashtag\"}],\"type\":\"Add\",\"actor\":\"https://mastodon.social/users/Gargron\",\"target\":\"https://mastodon.social/users/Gargron/collections/featured\",\"object\":{\"type\":\"Hashtag\",\"href\":\"https://mastodon.social/@Gargron/tagged/mastodev\",\"name\":\"#mastodev\"},\"signature\":{\"type\":\"RsaSignature2017\",\"creator\":\"https://mastodon.social/users/Gargron#main-key\",\"created\":\"2022-10-24T09:50:09Z\",\"signatureValue\":\"TLHntjiGqeTdET8OSA/gr1HGUU0x/Udeags/sPV0aTaH4R+eNY6Zj0Ejhx8wYm6GT+5clBl3JbVq+1G45cLMVoKUM66+J61oMkk6yN041n/4LRooV2OTbm5R7tRxn07T2etuyYzD1xVPRj2T+CyjjA2qWlQFC0MoYDKerYfSk8c6bwSABybfS/U77M9h1NFNgkoUFRFpF8PqHl3mY+W/MhbbfTqn07x+MUvPbMiVf7fCJSUuKPwHEMoysSLzMvzcUr+t/23X8MWQcanp2Tk6AXQ6qCwL5yxcPynAhRW1hWrzE/La52+E8HVvoAUB4TDbOVxSm1nps655j8oDQRunEQ==\"}}", nil, "Account"
```

Error backtrace:
```
/opt/mastodon/live/app/lib/activitypub/activity.rb:153:in `fetch_remote_original_status'
/opt/mastodon/live/app/lib/activitypub/activity.rb:113:in `status_from_object'
/opt/mastodon/live/app/lib/activitypub/activity/add.rb:7:in `perform'
```
https://github.com/mastodon/mastodon/blob/5fa340931e2bd39a4bcccc9a97ad58e13a2d56ab/app/lib/activitypub/activity.rb#L152-L159

### Specifications

3.5.3, latest

**Comment by shleeable** [comment]
I believe this will be resolved once you upgrade to 4.0

**Comment by ClearlyClaire** [comment]
Oh, seems like it's the recent addition of featured tags to the featured collection not being gracefully handled by released versions of Mastodon. I wonder what can be done to improve the situation.

**Comment by andrew-d** [comment]
This patch doesn't actually fetch anything, but it should prevent the Sidekiq workers from crashing and filling up the dead queue:
```diff
--- a/app/lib/activitypub/activity.rb
+++ b/app/lib/activitypub/activity.rb
@@ -150,7 +150,7 @@ class ActivityPub::Activity
   end
 
   def fetch_remote_original_status
-    if object_uri.start_with?('http')
+    if object_uri&.start_with?('http')
       return if ActivityPub::TagManager.instance.local_uri?(object_uri)
       ActivityPub::FetchRemoteStatusService.new.call(object_uri, id: true, on_behalf_of: @account.followers.local.first)
     elsif @object['url'].present?
```

`fetch_remote_original_status` is called from `status_from_object`, and all callers of `status_from_object` should handle the case where it returns `nil`.

**Comment by flaki** [comment]
> I believe this will be resolved once you upgrade to 4.0

I'm sorry but there's got to be a better answer to "the latest update of our fediverse server trivially broke backwards-compat with older versions, you should just upgrade" and a shrug.

**Comment by Gargron** [comment]
It adds errors to the logs for stuff that previously wasn’t federated but where does it break any compatibility?
